ndk/src/event.rs

```diff
@@ -47,12 +47,20 @@ pub enum Source {
     Any = ffi::AINPUT_SOURCE_ANY,
 }
 
+impl Source {
+    pub fn class(self) -> SourceClass {
+        let class = u32::from(self) & ffi::AINPUT_SOURCE_CLASS_MASK;
+        // TODO: Don't use None
+        class.try_into().unwrap_or(SourceClass::None)
```

Comment:
I'll merge it like this and clean this up in a followup PR that revises all `enum` types.